### PR TITLE
[24.10] RFC: backport changes from main and remove USE_APK config option

### PR DESCRIPTION
**Pull request by ynezz**
From https://lists.openwrt.org/pipermail/openwrt-devel/2024-November/043431.html 

> I would prefer if we rebase OpenWrt 24.10 branch on top of main again 
because we haven't done much testing on 24.10 specifically and there are 
not many bigger changes in main. Probably half of the changes done in 
the last month were already backported or are proposed as backports.

This pull request is my alternative to the proposed rebase, cherry-picking everything from main, but leaving any `apk` related changes as they might introduce regressions with `opkg`. It was decided, that `ipq60xx` is not ready for prime time yet, so it will need to wait for the next release.

>    * We should revert the setting making APK default and hide the option

Done in 9d2e5dc13bf894e2652d3e1c7a041d57e479dc29

### CODE REVIEW

File: config/Config-build.in [hunk 1]
```diff
@@ -68,10 +68,6 @@ menu "Global build settings"
 		bool "Enable TLS certificate verification during package download"
 		default y
 
-	config USE_APK
```
Comment (aparcar) [inline config/Config-build.in:71]: I'm in favor of this change for 24.10

**Comment by KanjiMonster** [comment]
> We should revert the setting making APK default and hide the option

Should the apk package also be marked as broken/removed/hidden? People might still try to install/add it assuming this makes it work.

**Review by GoetzGoerisch — APPROVED** [review]
~~Suggest renaming branch and release to 24.11, otherwise ok.~~
Approved from my pov.
Can follow @efahl s view for not renaming.

**Review by namiltd — APPROVED** [review]

**Comment by ynezz** [comment]
> > We should revert the setting making APK default and hide the option
> 
> Should the apk package also be marked as broken/removed/hidden? People might still try to install/add it assuming this makes it work.

Yes, good point.

**Review by hauke — APPROVED** [review]
LGTM

**Comment by hauke** [comment]
I think this should also go into 24.10. These problems were seen with apk, but I think they also exist with opkg
 * adb921c34ff220951a699665bb35dc0e3538b3b2 gdb: Fix description for apk, remove backtick
 * a4bd42b79015af251694a84bc2bbb7d42b34f47b config: dont select procd-seccomp via USE_SECCOMP
    * I am not sure, I haven't fully understood the problem

**Comment by robimarko** [comment]
> * [a4bd42b](https://github.com/openwrt/openwrt/commit/a4bd42b79015af251694a84bc2bbb7d42b34f47b) config: dont select procd-seccomp via USE_SECCOMP
>   
>   * I am not sure, I haven't fully understood the problem


Yes, this is a generic issue that was uncovered when SECCOMP selection was moved to base-files as a dependency, our buildroot seems to have some kind of a conflict in which CONFIG_SECCOMP is always set to y

**Comment by ynezz** [comment]
> * https://github.com/openwrt/openwrt/commit/adb921c34ff220951a699665bb35dc0e3538b3b2 gdb: Fix description for apk, remove backtick

Good point, I avoided everything with `apk` in the name.

> * https://github.com/openwrt/openwrt/commit/a4bd42b79015af251694a84bc2bbb7d42b34f47b config: dont select procd-seccomp via USE_SECCOMP

I avoided this one, because it  started appearing after apk changes, which we're not backporting and root of the problem is not know (I run out of ideas/time when looking for it initially), so don't have complete picture to judge it.

> I prefer this commit: https://github.com/hauke/openwrt/commit/2413d25272b96046f727e2426f0084ae1c915269

Ok. @hauke what about the apk itself https://github.com/openwrt/openwrt/pull/17097/commits/4ecd582755ec9333be58de83f229646045762d67? Mark it as broken as well?

**Pull request merged**